File: python/DS/Trie.py
class TrieNode:
    def __init__(self):
        self.children = [None]*26
        self.endOfWord = False
    
class Trie:
    def __init__(self):
        self.root = self.getNode()
    
    def getNode(self):
        return TrieNode()
    
    def _charToIndex(self, ch):
        return ord(ch) - ord('a')
    
    def insert(self, key):
        pCrawl = self.root
        length = len(key)
        for level in range(length):
            index = self._charToIndex(key[level])
            if not pCrawl.children[index]:
                pCrawl.children[index] = self.getNode()
            pCrawl = pCrawl.children[index]
        
        pCrawl.endOfWord = True
    
    def search(self, key):
        pCrawl = self.root
        length = len(key)
        for level in range(length):
            index = self._charToIndex(key[level])
            if not pCrawl.children[index]:
                return False
            pCrawl = pCrawl.children[index]
        
        return pCrawl != None and pCrawl.endOfWord
    
keys = ["the","a","there","anaswe","any", "by","their"] 

t = Trie() 
 
for key in keys: 
    t.insert(key)

print(t.search('anaswe'))

File: java/ds/Trie.java
public class Trie {

    static final int ALPHABET_SIZE = 26;
    TrieNode root;
    public Trie() {
        root = new TrieNode();
    }
    
    class TrieNode {
        TrieNode children[];
        boolean isEndOfWord;

        TrieNode() {
            children = new TrieNode[ALPHABET_SIZE];
            for(int i=0; i<ALPHABET_SIZE; ++i) {
                children[i] = null;
            }
            isEndOfWord = false;
        }
    }

    public void insert(String key) {
        int length = key.length();
        TrieNode pCrawl = root;
        int index;

        for(int i=0; i<length; ++i) {
            index = key.charAt(i) - 'a';
            if(pCrawl.children[index] == null)
                pCrawl.children[index] = new TrieNode();
            pCrawl = pCrawl.children[index];
        }

        pCrawl.isEndOfWord = true;
    }

    public boolean search(String key) {
        int length = key.length();
        TrieNode pCrawl = root;
        int index;

        for(int i=0; i<length; ++i) {
            index = key.charAt(i) - 'a';
            if(pCrawl.children[index] == null)
                return false;
            pCrawl = pCrawl.children[index];
        }
        return pCrawl != null && pCrawl.isEndOfWord;
    }

    public static void main(String[] args) {
        Trie myTrie = new Trie();
        String keys[] = {"my", "name", "is", "shaunak"};
        for(int i=0; i<keys.length; ++i) {
            myTrie.insert(keys[i]);
        }
        for(int i=0; i<keys.length; ++i) {
            System.out.println(myTrie.search(keys[i]));
        }
    }


}